cleanlab/internal/util.py

```diff
@@ -626,7 +662,7 @@ def is_tensorflow_dataset(X) -> bool:
     return False  # assumes this cannot be tensorflow dataset if tensorflow cannot be imported
 
 
-def csr_vstack(a, b) -> DatasetLike:
+def csr_vstack(a: NDArray, b: NDArray) -> DatasetLike:
```

Comment:
CSR matrices are not numpy arrays, because they use a different storage format (3 vectors) for the non-zero values in the matrix.

You may want to look into https://docs.scipy.org/doc/scipy/reference/sparse.html.